src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs

```diff
@@ -40,15 +41,18 @@ public async Task Try()
                 if (sub.Collection == null)
                     throw new ArgumentException("Collection must be specified");
 
+                const int maxPageSize = 1024;
                 var pageSize = GetIntValueQueryString("pageSize") ?? 1;
+                if (pageSize > maxPageSize)
```

Comment:
Why have this limit? We already have a time limit that also applies, no?

Reply:
I do not want users to request 1M results here and kill the Studio. This is used only for Studio/Subscription/Test functionality and the 1024 seams more than fine?

Reply:
Why bother? This isn't happening on the studio, it happens on the server, and limited by the time.
It is actually going to be pretty important to have large numbers there because we may need to scan a lot of documents to find a useful value if there is a where there.

Reply:
This is not related to the number of documents to scan (it is int.Max). This is a number of results to return to the user in Studio and display in the grid there. We can remove this, but I think 1024 is more than enough?

Reply:
Oh, number of docs to return is fine, then.